### PR TITLE
feat: port Data.Rbtree.Init

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/RBTree/Init.lean [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#align rbnode.depth Std.RBNode.depth
+#align rbnode.min Std.RBNode.min
+#align rbnode.max Std.RBNode.max
+#align rbnode.fold Std.RBNode.fold
+
+#noalign rbnode.rev_fold
+
+#align rbnode.balance1 Std.RBNode.balance1
+
+#noalign rbnode.balance1_node
```
Comment (ChrisHughes24) [inline Mathlib/Data/RBTree/Init.lean:38]: Are these things `noalign` for a reason? We don't want to port them?

Comment (jcommelin) [reply Mathlib/Data/RBTree/Init.lean:38]: cc @digama0

Comment (digama0) [reply Mathlib/Data/RBTree/Init.lean:38]: These align statements are for documentation purposes only, they are significantly different in the details. I'm not sure going over all the individual technical lemmas will be that helpful. From a cursory look it seems that `rbnode.balance1_node` is more similar to `Std.RBNode.balance1` than `rbnode.balance1`, so you could align both of them to `Std.RBNode.balance1`? :shrug:

**Comment by semorrison** [comment]
Made obsolete by #5504, which incorporated these #aligns.

**Pull request closed without merging**